pennylane/pauli/conversion.py

```diff
@@ -220,7 +129,212 @@ def pauli_decompose(
+
+    Args:
+        H (array[complex]): a Hermitian matrix of dimension :math:`2^n\times 2^n`.
+        hide_identity (bool): does not include the Identity observable within
+            the tensor products of the decomposition if ``True``.
+        wire_order (list[Union[int, str]]): the ordered list of wires with respect
+            to which the operator is represented as a matrix.
+        pauli (bool): return a PauliSentence instance if ``True``.
+        check_hermitian (bool): check if the provided matrix is Hermitian if ``True``.
+
+    Returns:
+        Union[~.Hamiltonian, ~.PauliSentence]: the matrix decomposed as a linear combination
+        of Pauli operators, either as a :class:`~.Hamiltonian` or :class:`~.PauliSentence` instance.
+
+    .. seealso::
+
+        :func:`~.pauli_decompose_with_phase` for decomposing any general matrix.
+
+    **Example:**
+
+    We can use this function to compute the Pauli operator decomposition of an arbitrary Hermitian
+    matrix:
+
+    >>> A = np.array(
+    ... [[-2, -2+1j, -2, -2], [-2-1j,  0,  0, -1], [-2,  0, -2, -1], [-2, -1, -1,  0]])
+    >>> H = qml.pauli_decompose(A)
+    >>> print(H)
+    (-1.5) [I0 X1]
+    + (-1.5) [X0 I1]
+    + (-1.0) [I0 I1]
+    + (-1.0) [I0 Z1]
+    + (-1.0) [X0 X1]
+    + (-0.5) [I0 Y1]
+    + (-0.5) [X0 Z1]
+    + (-0.5) [Z0 X1]
+    + (-0.5) [Z0 Y1]
+    + (1.0) [Y0 Y1]
+
+    We can return a :class:`~.PauliSentence` instance by using the keyword argument ``pauli=True``:
+
+    >>> ps = qml.pauli_decompose(A, pauli=True)
+    >>> print(ps)
+    -1.0 * I
+    + -1.5 * X(1)
+    + -0.5 * Y(1)
+    + -1.0 * Z(1)
+    + -1.5 * X(0)
+    + -1.0 * X(0) @ X(1)
+    + -0.5 * X(0) @ Z(1)
+    + 1.0 * Y(0) @ Y(1)
+    + -0.5 * Z(0) @ X(1)
+    + -0.5 * Z(0) @ Y(1)
+
+    We can also set custom wires using the ``wire_order`` argument:
+
+    >>> ps = qml.pauli_decompose(A, pauli=True, wire_order=['a', 'b'])
+    >>> print(ps)
+    -1.0 * I
+    + -1.5 * X(b)
+    + -0.5 * Y(b)
+    + -1.0 * Z(b)
+    + -1.5 * X(a)
+    + -1.0 * X(a) @ X(b)
+    + -0.5 * X(a) @ Z(b)
+    + 1.0 * Y(a) @ Y(b)
+    + -0.5 * Z(a) @ X(b)
+    + -0.5 * Z(a) @ Y(b)
+    """
+    n = int(qml.math.log2(len(H)))
+    N = 2**n
+
+    if H.shape != (N, N):
+        raise ValueError("The matrix should have shape (2**n, 2**n), for any qubit number n>=1")
+
+    if check_hermitian and not qml.math.allclose(H, qml.math.conjugate(qml.math.transpose(H))):
+        raise ValueError("The matrix is not Hermitian")
+
+    coeffs, obs = _generalized_pauli_decompose(
+        H, hide_identity=hide_identity, wire_order=wire_order, pauli=pauli, padding=False
+    )
+    coeffs = qml.math.real(coeffs)
+
+    if pauli:
+        return PauliSentence(
+            {
+                PauliWord({w: o for o, w in obs_n_wires}): coeff
+                for coeff, obs_n_wires in zip(coeffs, obs)
+            }
+        )
+
+    with qml.QueuingManager.stop_recording():
```

Comment:
nice 🤝 buuut this wasn't there before. is this considered a breaking change (that calling `pauli_decompose` doesn't queue the returned Hamiltonian)? I suppose they don't really queue on their own anyway (need to be assigned to a MeasurementProcess), so I wonder what motivated adding this.

Reply:
Yes, I do think they weren't getting queued when I was doing stuff in qnode, but were getting queued when working with `qml.queuing.AnnotatedQueue()`. I thought we didn't want it so I added this. Let me know what is supposed to be the expected behaviour. 🤔

Reply:
not super necessary but whatever you prefer. they are "queued" in the sense that they get added to the `AnnotatedQueue` object. They are removed in the processing of that queue (eg. `QuantumScript.from_queue(q)`) unless wrapped in some MeasurementProcess (`Hamiltonian` has no "queue category" so it doesn't get added to the tape, whereas `expval(Hamiltonian)` or any observable has the measurement queue category).